### ISSUE TITLE
Is the one-prediction-per-sequence variant still forthcoming?

### ISSUE THREAD

**Issue by joecomerisnotavailable** [mp2893/gram#7]
Or, would it be possible to give a rough outline of where the major changes would need to be made in the current code to convert it?

Thanks

**Comment by mp2893** [comment]
Since Theano is no longer maintained, I don't plan to add anything to this repo.
But changing the current code to one-pred-per-seq version shouldn't be that hard.

Below are some things that need to be done in gram.py:
- line 81-98: use line 56-74 of [rnn_predict](https://github.com/mp2893/rnn_predict/blob/master/gru_emb.py).
- line 119: y should be either a matrix (in case of multi-class prediction) or a vector (in case of binary prediction), not a 3D tensor, because you make only one prediction per sequence.
- line 121: you don't need lengths anymore. Be sure to delete lengths in everywhere in the code.
-line 135-147: see line 89-98 of [rnn_predict](https://github.com/mp2893/rnn_predict/blob/master/gru_emb.py).
- line 149, load_data(): Load an appropriate label file.
- line 224-240: you don't need a 3d tensor y. Prepare an appropriate y.

Basically those are the high-level directions. [rnn_predict](https://github.com/mp2893/rnn_predict/blob/master/gru_emb.py) should give you a pretty good idea how to modify gram.py for binary prediction.

Best,
Ed
